dashboard/app/models/concerns/partial_registration.rb

```diff
@@ -35,6 +35,7 @@ def self.persist_attributes(session, user)
   end
 
   def self.delete(session)
+    return unless in_progress? session
```

Comment:
worth adding a comment here explaining the reason behind this guard clause? It's not immediately obvious that what we're trying to do here is avoid a `nil` argument

Reply:
Done!